crates/oxc_linter/src/rules/unicorn/prefer_string_starts_ends_with.rs

```diff
@@ -92,7 +92,10 @@ enum ErrorKind {
 }
 
 fn check_regex(regexp_lit: &RegExpLiteral) -> Option<ErrorKind> {
```

Comment:
there are perf optimizations: e.g. we loop over the whole regex 2x: once in `is_simple_string` and 1x in `is_useless_case_sensitive_regex_flag`. but i think the perf gains are negligible

Reply:
Yes stats on this case is rare.